File: gm-utils/js/src/commom/utils/string/StringSplit.js
import ArrayLst from '../array/ArrayLst';
import Null from '../object/Null';

export default class StringSplit {

	static exec(s, separator) {

		if (Null.is(s)) {
			return new ArrayLst();
		}

		if (s.indexOf(separator) === -1) {
			return ArrayLst.build(s);
		}

		if (separator.length === 0) {

			let lst = new ArrayLst();

			while (s.length > 0) {
				lst.add(s.substring(0,1));
				s = s.substring(1);
			}

			return lst;

		}

		let sp = separator;

		let array = s.split(sp);

		let list = new ArrayLst();
		list.addArray(array);
		return list;

	}

}

File: gm-utils/src/main/java/src/commom/utils/string/StringSplit.java
package src.commom.utils.string;

import gm.languages.ts.javaToTs.JS;
import gm.utils.anotacoes.IgnorarDaquiPraBaixo;
import src.commom.utils.array.Itens;
import src.commom.utils.object.Null;

public class StringSplit extends JS {

	public static Itens<String> exec(String s, String sub) {
		
		Itens<String> itens = new Itens<>();
		
		if (Null.is(s)) {
			return itens;
		}
		
		if (Null.is(sub)) {
			sub = "";
		}
		
		itens.addArray(split(s, sub));
		
		return itens;
		
	}

	@IgnorarDaquiPraBaixo
	
	private StringSplit() {}
	
	public static void main(String[] args) {
		exec("/v1/clients/","/").print();
	}
	
}
